app/gimlet/rev-d.toml

```diff
@@ -201,14 +201,14 @@ notifications = ["timer"]
 
 [tasks.host_sp_comms]
 name = "task-host-sp-comms"
-features = ["stm32h753", "uart7", "baud_rate_3M", "hardware_flow_control", "vlan"]
+features = ["stm32h753", "uart7", "baud_rate_3M", "hardware_flow_control", "vlan", "gimlet"]
 uses = ["uart7"]
 interrupts = {"uart7.irq" = "usart-irq"}
 priority = 7
-max-sizes = {flash = 32768, ram = 32768}
-stacksize = 2048
+max-sizes = {flash = 65536, ram = 32768}
+stacksize = 6000
```

Comment:
This is a pretty big jump. Do you know where the extra size is coming from?

Reply:
Yeah, I had the same reaction.  I think it's that the `InventoryData` type is ~512 bytes (since it holds a full DIMM SPD), and there are a few stack frames that aren't being inlined since we're using lambda functions here.

I knocked about 2K off it by allocating the `InventoryData` once in the outer frame, then using references to it in the lambda functions.

| situation | stack depth |
|-|-|
| baseline |       864 |
| reading DIMM |   1672 |
|reading from packrat | 1776 |
|reading BRM491 | 1768 |
|reading AT24 barcode | 2712 |

I'm mystified as to why reading the AT24 barcode adds a whole 1K of stack depth.  I'm going to look at [`read_at24csw080_id`](https://github.com/oxidecomputer/hubris/blob/7316fe07f5a7f17a5339584a4f178fbe462d9de3/task/host-sp-comms/src/bsp/gimlet_bcd.rs#L512-L534) a little more and see if I can figure it out, but more eyes are welcome here.

Reply:
Aha: `read_at24csw080` wasn't being inlined, so we had one stack frame in `perform_inventory_lookup` and another in `read_at24csw080_id`, costing us an extra 512 bytes.  Explicitly allocating the data in the caller and passing it by reference brings us down to 2200, which seems reasonable.